bundles/auth/core/src/main/java/org/apache/sling/auth/core/impl/SlingAuthenticator.java

```diff
@@ -1639,4 +1639,35 @@ protected AbstractAuthenticationHandlerHolder createHolder(String path,
         }
     }
 
+    /**
+     * SLING-6053
+     * Auth handler is required for the node and for all his children.
+     * Solved the case when sibling node name startsWith same name.
+     * <p>
+     * Eg: node: /page & node /page1
+     */
+    static class AuthenticationHandlerPath {
```

Comment:
Is a simple private method just taking two strings in the SlingAuthenticator not enough?